fendermint/vm/topdown/src/sync.rs

```diff
@@ -15,19 +15,55 @@ use ipc_sdk::subnet_id::SubnetID;
 use std::sync::Arc;
 use std::time::Duration;
 
+/// Query the parent finality from the block chain state
+pub trait ParentFinalityStateQuery {
+    /// Get the latest committed finality from the state
+    fn get_latest_committed_finality(&self) -> anyhow::Result<Option<IPCParentFinality>>;
+}
+
 /// Constantly syncing with parent through polling
-pub struct PollingParentSyncer {
+pub struct PollingParentSyncer<P> {
     config: Config,
-    parent_view_provider: Arc<InMemoryFinalityProvider>,
+    parent_view_provider: Arc<P>,
     agent: Arc<IPCAgentProxy>,
 }
 
-impl PollingParentSyncer {
-    pub fn new(
-        config: Config,
-        parent_view_provider: Arc<InMemoryFinalityProvider>,
-        agent: Arc<IPCAgentProxy>,
-    ) -> Self {
+/// Start the polling parent syncer in the background
+pub async fn launch_polling_syncer<
+    Q: ParentFinalityStateQuery + Send + Sync + 'static,
+    P: ParentFinalityProvider + Send + Sync + 'static,
+>(
+    query: &Q,
+    config: Config,
+    view_provider: Arc<P>,
+    agent: IPCAgentProxy,
+) -> anyhow::Result<()> {
+    loop {
+        let finality = match query.get_latest_committed_finality() {
+            Ok(Some(finality)) => finality,
+            Ok(None) => {
+                tracing::debug!("app not ready for query yet");
+                tokio::time::sleep(Duration::from_secs(5)).await;
+                continue;
+            }
+            Err(e) => {
+                tracing::warn!("cannot get committed finality: {e}");
+                tokio::time::sleep(Duration::from_secs(5)).await;
+                continue;
+            }
+        };
+
+        atomically_or_err(|| view_provider.set_new_finality(finality.clone())).await?;
```

Comment:
I feel adding the height into config should do it, but still think we need the initialisation process. Because in Step 2, `Create an already initialized provider from it.`, one still has to wait to read from the ledger to confirm what's the starting height. This is an initialisation process.